### PR TITLE
GH-9738: Restricting access to approval dashboard to unauthorized users

### LINKED ISSUE #9738: Unauthorized users that try to access the approval dashboard should be redirected to the home page
When this is fixed, please remember to unskip the corresponding test case on `order-approval.e2e-spec.ts`

### PR DESCRIPTION
**Pull request by plabadie**
closes GH-9738

### CODE REVIEW

File: feature-libs/organization/order-approval/components/details/order-approval-details.module.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ import {
   OrderDetailTotalsComponent,
   SpinnerModule,
 } from '@spartacus/storefront';
+import { ApproverGuard } from '../../core/public_api';
```
Comment (bgambocjaviniar) [inline feature-libs/organization/order-approval/components/details/order-approval-details.module.ts]: instead of going to the public_api through relative path to access Approver guard, I believe there's a pattern set for these `feature-libs`. 

But it would be to update the tsconfig*.json to be able to use the suggested fix. Moreover, the current implementation has import errors that is causing the build to fail and I wasn't able to run the project manually. 

I believe it would be best to follow how to import is being used like how `AdminGuard` does it when they import it to certain configs for usage
```suggestion
import { ApproverGuard } from '@spartacus/organization/order-approval/core';
```

**Review by bgambocjaviniar — CHANGES_REQUESTED** [review]

File: feature-libs/organization/order-approval/components/details/order-approval-details.module.ts (hunk 1 above)
Comment (plabadie) [reply feature-libs/organization/order-approval/components/details/order-approval-details.module.ts]: Good catch!

That import was wrong.  However we import the guard using a relative path because the approvals feature does not have a standalone "core" entry point.  Just root, asstes, and another one for the rest.

**Review by bgambocjaviniar — CHANGES_REQUESTED** [review]
LGTM
- anonymous users gets sent back to login
- unauth users gets sent back to homepage 
- clean stuff :) 

However,
getting consistent failure from the test runner and headless of the e2e test, which is failing starting from `it('should NOT display order approval option in menu', () => {`

Same reasoning here https://github.com/SAP/spartacus/pull/9835#pullrequestreview-535178558 

Basically, `findByText` causes flakiness from my past observations. I would stop using that chainer and use other ones like .should or .contain. In this case, it is specifically failing because of the function that uses `findByText` in the helper of b2b-order-approval's helper method `checkApprovalDashboardMenuOptionExistence`.

I would change 

```
cy.get('cx-navigation > cx-navigation-ui').within(() => {
    cy.get('nav > cx-generic-link')
      **.findByText('Approval Dashboard')**
      .should(exists ? 'exist' : 'not.exist');
  });
```

to

```
cy.get('cx-navigation > cx-navigation-ui').within(() => {
    cy.get('nav > cx-generic-link')
      **.contains('Approval Dashboard')**
      .should(exists ? 'exist' : 'not.exist');
  });
```

In conclusion, some days `findByText` passes, but other days, it doesn't, which causes these flakiness at times. 
HOWEVER, I am kind of confused now since `checkApprovalDashboardMenuOptionExistence` is being used before this assertion elsewhere and when it reaches this one, it fails.

---
what do you think?

**Review by bgambocjaviniar — CHANGES_REQUESTED** [review]
I believe this was missed https://github.com/SAP/spartacus/pull/9834#pullrequestreview-535675713

Screen Shot 2020-11-20 at 3 36 34 PM

**Review by bgambocjaviniar — APPROVED** [review]
🟢

**Comment by plabadie** [comment]
replaced by `.contains` to be on the safe side.  However the issue only appears on cypress 4.x and not on cypress 5 which we upgraded to.
